test/core/security/grpc_tls_certificate_provider_test.cc

```diff
@@ -495,6 +495,43 @@ TEST_F(GrpcTlsCertificateProviderTest,
   CancelWatch(watcher_state_1);
 }
 
```

Comment:
How about we also add a test, where the initial credentials are invalid?

Reply:
working on that. I am will be talking to Zhen to make sure I implement the desired behavior